vary.js

```diff
@@ -1,16 +1,65 @@
 'use strict'
 
-const append = require('vary').append
+const LRU = require('tiny-lru')
+const regexCache = new LRU(1000)
+
+/**
+ * RegExp to match field-name in RFC 7230 sec 3.2
+ *
+ * field-name    = token
+ * token         = 1*tchar
+ * tchar         = "!" / "#" / "$" / "%" / "&" / "'" / "*"
+ *               / "+" / "-" / "." / "^" / "_" / "`" / "|" / "~"
+ *               / DIGIT / ALPHA
+ *               ; any VCHAR, except delimiters
+ */
+
+const fieldnameChars = '!#$%&\'*+-.^_`|~0-9A-Za-z'
+const fieldNameRE = new RegExp(`^[${fieldnameChars}]+$`)
+const wildcardRE = fieldRegex('*', '')
+
+function escapeRegex (value) {
+  return value.replace(/[$*+-.^|]/g, '\\$&')
+}
+
+function fieldRegex (field, flags = 'i') {
+  if (fieldNameRE.test(field) === false) {
+    throw new TypeError('field argument contains an invalid header name')
+  }
+  const escapedField = escapeRegex(field)
+  return new RegExp(`(^|(.*,))[^${fieldnameChars}]*${escapedField}[^${fieldnameChars}]*(,.*|$)`, flags)
+}
```

Comment:
How much of this code is extracted from the vary module?

Reply:
fieldNameRE is from the vary module and the TypeError is transplanted from vary to ensure that it is backwards compatible. Other than that nothing.

Reply:
Please add the license disclaimer of that module in this file

Reply:
The code snippet I copied is basically 1:1 copied from the RFC. So I referenced the RFC. I changed Error-Messages.

So basically there is no copyrighted material in my PR, as it is based on the RFC.